apache2/re.c

```diff
@@ -1970,6 +1970,48 @@ msre_ruleset *msre_ruleset_create(msre_engine *engine, apr_pool_t *mp) {
 
     return ruleset;
 }
+/**
+
+
+ * Creates a ruleset by copy the parent ruleset.
+ *
+ * @param mp apr pool structure
+ * @param parent_ruleset Parent's msre_ruleset
+ *
+ * return the address of created ruleset if OK.
+ * return NULL if Something went wrong.
+ */
+
+msre_ruleset *msre_ruleset_create_and_copy(msre_engine *engine, apr_pool_t *mp, msre_ruleset *parent_ruleset) {
+    msre_ruleset *child_ruleset;
+
+    child_ruleset = apr_pcalloc(mp, sizeof(msre_ruleset));
+    if (child_ruleset == NULL) return NULL;
```

Comment:
what will happen to the calling code if this happen? Will we silently continue, will we crash badly (pull down entire w3wp?), or will we crash in a controlled way (500, etc.)

Reply:
The original code will silently continue and crash doesn't happen because other code will check the ruleset and skip it. It's the same as no rules in the modsecurity.